core/arch/arm/plat-imx/pm/pm-imx7.c

```diff
@@ -0,0 +1,264 @@
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <arm.h>
+#include <arm32.h>
+#include <console.h>
+#include <io.h>
+#include <imx.h>
+#include <imx_pm.h>
+#include <kernel/panic.h>
+#include <kernel/pm_stubs.h>
+#include <kernel/cache_helpers.h>
+#include <mm/core_mmu.h>
+#include <mm/core_memprot.h>
+#include <mmdc.h>
+#include <platform_config.h>
+#include <sm/psci.h>
+#include <sm/sm.h>
+#include <sm/pm.h>
+#include <string.h>
+
+paddr_t iram_tlb_phys_addr = -1UL;
+void *iram_tlb_virt_addr;
+
+#define READ_DATA_FROM_HARDWARE		0
+
+static uint32_t imx7d_ddrc_ddr3_setting[][2] = {
+	{ 0x0, READ_DATA_FROM_HARDWARE },
+	{ 0x1a0, READ_DATA_FROM_HARDWARE },
+	{ 0x1a4, READ_DATA_FROM_HARDWARE },
+	{ 0x1a8, READ_DATA_FROM_HARDWARE },
+	{ 0x64, READ_DATA_FROM_HARDWARE },
+	{ 0x490, READ_DATA_FROM_HARDWARE },
+	{ 0xd0, READ_DATA_FROM_HARDWARE },
+	{ 0xd4, READ_DATA_FROM_HARDWARE },
+	{ 0xdc, READ_DATA_FROM_HARDWARE },
+	{ 0xe0, READ_DATA_FROM_HARDWARE },
+	{ 0xe4, READ_DATA_FROM_HARDWARE },
+	{ 0xf4, READ_DATA_FROM_HARDWARE },
+	{ 0x100, READ_DATA_FROM_HARDWARE },
+	{ 0x104, READ_DATA_FROM_HARDWARE },
+	{ 0x108, READ_DATA_FROM_HARDWARE },
+	{ 0x10c, READ_DATA_FROM_HARDWARE },
+	{ 0x110, READ_DATA_FROM_HARDWARE },
+	{ 0x114, READ_DATA_FROM_HARDWARE },
+	{ 0x120, READ_DATA_FROM_HARDWARE },
+	{ 0x180, READ_DATA_FROM_HARDWARE },
+	{ 0x190, READ_DATA_FROM_HARDWARE },
+	{ 0x194, READ_DATA_FROM_HARDWARE },
+	{ 0x200, READ_DATA_FROM_HARDWARE },
+	{ 0x204, READ_DATA_FROM_HARDWARE },
+	{ 0x214, READ_DATA_FROM_HARDWARE },
+	{ 0x218, READ_DATA_FROM_HARDWARE },
+	{ 0x240, READ_DATA_FROM_HARDWARE },
+	{ 0x244, READ_DATA_FROM_HARDWARE },
+};
+
+static uint32_t imx7d_ddrc_phy_ddr3_setting[][2] = {
+	{ 0x0, READ_DATA_FROM_HARDWARE },
+	{ 0x4, READ_DATA_FROM_HARDWARE },
+	{ 0x10, READ_DATA_FROM_HARDWARE },
+	{ 0xb0, READ_DATA_FROM_HARDWARE },
+	{ 0x9c, READ_DATA_FROM_HARDWARE },
+	{ 0x7c, READ_DATA_FROM_HARDWARE },
+	{ 0x80, READ_DATA_FROM_HARDWARE },
+	{ 0x84, READ_DATA_FROM_HARDWARE },
+	{ 0x88, READ_DATA_FROM_HARDWARE },
+	{ 0x6c, READ_DATA_FROM_HARDWARE },
+	{ 0x20, READ_DATA_FROM_HARDWARE },
+	{ 0x30, READ_DATA_FROM_HARDWARE },
+	{ 0x50, 0x01000010 },
+	{ 0x50, 0x00000010 },
+	{ 0xc0, 0x0e407304 },
+	{ 0xc0, 0x0e447304 },
+	{ 0xc0, 0x0e447306 },
+	{ 0xc0, 0x0e447304 },
+	{ 0xc0, 0x0e407306 },
+};
+
+static struct imx7_pm_data imx7d_pm_data_ddr3 = {
+	.ddrc_num = ARRAY_SIZE(imx7d_ddrc_ddr3_setting),
+	.ddrc_offset = imx7d_ddrc_ddr3_setting,
+	.ddrc_phy_num = ARRAY_SIZE(imx7d_ddrc_phy_ddr3_setting),
+	.ddrc_phy_offset = imx7d_ddrc_phy_ddr3_setting,
+};
+
+paddr_t phys_addr[] = {
+	AIPS1_BASE, AIPS2_BASE, AIPS3_BASE
+};
+
+int pm_imx7_iram_tlb_init(void)
+{
+	uint32_t i;
+	struct tee_mmap_region map;
+
+	/* iram tlb already initialized */
```

Comment:
This is to build mmu table for suspend. When code run in iram, ddr is not accessible here.

Reply:
Agree. It's the three letters in "tlb", how did you choose them?

Reply:
sorry, @jenswi-linaro , I could not follow clearly. Do you mean "tlb" is not a good name here?

Reply:
tlb for me is usually Translation Lookaside Buffer, but given a good explanation I'd accept it in this context too.

Or is it just a spell error that has been copied, was it supposed to be "tbl" from the beginning?

Reply:
`tbl` sounds more sane to me based on the discussion between the two of you. Also when it's in written format in the comment (not function names etc) why not even just write "table" instead of an abbreviation?

Reply:
@jenswi-linaro @jbech-linaro "tbl" is better here, translation table. I did not find a good name, "mmu_table" too long, so picked "tlb" which caused misunderstanding. I'll replace tlb with tbl in this patch includeing functions and variables and add more comments.